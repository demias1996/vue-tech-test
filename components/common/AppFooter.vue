<template>
    <footer class="app-footer container d-flex align-items-center justify-content-center">
        <b-icon-bookmarks-fill class="mr-2" />
        <p class="footer-title mb-0">{{ title }}</p>
    </footer>
</template>

<script>
    import { BIconBookmarksFill } from 'bootstrap-vue';

    export default {
        name: 'app-footer',

        components: {
            BIconBookmarksFill,
        },

        props: {
            title: {
                type: String,
                default: '',
            },
        },
    };
</script>

<style lang="scss" scoped>
    .app-footer {
        min-height: 70px;
    }
</style>
