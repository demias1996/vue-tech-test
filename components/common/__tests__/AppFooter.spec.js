import { shallowMount } from '@vue/test-utils';
import AppFooter from '@/components/common/AppFooter';

describe('AppFooter', () => {
    test('is a Vue instance', () => {
        const wrapper = shallowMount(AppFooter);
        expect(wrapper.vm).toBeTruthy();
    });

    test('is rendered correctly with default props', () => {
        const wrapper = shallowMount(AppFooter);
        expect(wrapper.element).toMatchSnapshot();
    });

    test('title to be displayed', () => {
        const wrapper = shallowMount(AppFooter, {
            propsData: {
                title: 'Test title',
            },
        });

        const footerTitle = wrapper.find('.footer-title');

        expect(footerTitle.text()).toBe('Test title');
        expect(wrapper.element).toMatchSnapshot();
    });
});
