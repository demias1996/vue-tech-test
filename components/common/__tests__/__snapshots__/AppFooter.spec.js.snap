// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter is rendered correctly with default props 1`] = `
<footer
  class="app-footer container d-flex align-items-center justify-content-center"
>
  <b-icon-bookmarks-fill-stub
    class="mr-2"
    fontscale="1"
    rotate="0"
    scale="1"
    shifth="0"
    shiftv="0"
  />
   
  <p
    class="footer-title mb-0"
  >
    
  </p>
</footer>
`;

exports[`AppFooter title to be displayed 1`] = `
<footer
  class="app-footer container d-flex align-items-center justify-content-center"
>
  <b-icon-bookmarks-fill-stub
    class="mr-2"
    fontscale="1"
    rotate="0"
    scale="1"
    shifth="0"
    shiftv="0"
  />
   
  <p
    class="footer-title mb-0"
  >
    Test title
  </p>
</footer>
`;
